### ISSUE TITLE
After fine-tuning QwQ-32B, the performance becomes worse

### ISSUE THREAD

**Issue by yin-zhang** [GAIR-NLP/LIMO#36]
I used the training scripts to fine tune the QwQ-32B model.  However, after fine-tuning, the AIME24 performance drops from ~80% down to 60%。Is this expected?  Does it mean a strong baseline model gets overfitted to the small dataset, and loses part of its original inference capability?

The performance of Qwen2.5-32B-Instruct after finetuning does match the accuracy reported in the paper.  So my training setup should be correct.

**Comment by Yeez-lee** [comment]
> I used the training scripts to fine tune the QwQ-32B model. However, after fine-tuning, the AIME24 performance drops from ~80% down to 60%。Is this expected? Does it mean a strong baseline model gets overfitted to the small dataset, and loses part of its original inference capability?
> 
> The performance of Qwen2.5-32B-Instruct after finetuning does match the accuracy reported in the paper. So my training setup should be correct.

Maybe lower learning rate. QwQ-32B model is a strong reasoning model considering the size of 32B. Such compact data might lead to very slight performance improvements or even poorer performance if you are away from the optimal point. This dataset is more helpful for non-reasoning model to reasoning model, not reasoning one to a better model. Similar phenomenon is observed here (https://github.com/open-thoughts/open-thoughts/issues/26)

**Comment by HuangZhen02** [comment]
Thank you for raising this important point and sharing your findings.

The core issue is that QwQ-32B is already an exceptionally strong reasoning model. It is also a much more recent model compared to when the LIMO dataset was released. As a result, the quality of its pre-training data and its inherent reasoning capabilities likely already meet or exceed the complexity and scope of the LIMO dataset.

When you fine-tune such a powerful model on a dataset that does not introduce sufficiently novel or challenging reasoning tasks, the model can "overfit" to the characteristics of the smaller dataset. This process can inadvertently cause it to lose some of the more general, and powerful, inference abilities it originally possessed.
